sky/backends/cloud_vm_ray_backend.py

```diff
@@ -284,12 +284,13 @@ def check_ip():
                 ])
                 print('INFO: Reserved IPs:', gang_scheduling_id_to_ip)
 
-                if {cluster_ips_sorted!r} is not None:
-                    cluster_ips_map = {{ip: i for i, ip in enumerate({cluster_ips_sorted!r})}}
+                if {cluster_internal_ips_sorted!r} is not None:
```

Comment:
It seems like we should pass this in for `_execute_task_one_node()` as well, in which case we can remove Optional for this arg.

Reply:
Agreed! That would make reasoning about this a lot easier.